dpppt-backend-sdk/dpppt-backend-sdk-model/src/main/java/org/dpppt/backend/sdk/utils/UTCInstant.java

```diff
@@ -122,6 +122,26 @@ public LocalTime getLocalTime() {
     return getLocalDateTime().toLocalTime();
   }
 
+  /**
+   * @param releaseBucketDuration
+   * @return the start of the bucket this UTCInstant is in. If the UTCInstant is already at the
+   *     start of a bucket, it will be returned as-is.
+   */
+  public UTCInstant roundToBucketStart(Duration releaseBucketDuration) {
+    var rounding = releaseBucketDuration.toMillis();
+    return new UTCInstant((this.timestamp / rounding) * rounding);
+  }
+
+  /**
+   * @param releaseBucketDuration
+   * @return the start of the next bucket of this UTCInstant. If the UTCInstant is at the beginning
+   *     of a bucket, the next bucket will be returned.
+   */
+  public UTCInstant roundToNextBucket(Duration releaseBucketDuration) {
+    var rounding = releaseBucketDuration.toMillis();
+    return new UTCInstant(((this.timestamp / rounding) + 1) * rounding);
```

Comment:
I think this is dangerous for two reasons:

1) it relies on the correct type inference of `rounding` since it is `var`

2) it relies on integer division and no automatic type casts. Justs to give an example: Compare the output of `1/3` in a python2 interpreter and in a python3 (tested with 3.8.5) interpreter. Since they changed the API internally the latter is 0 and the former 0.333. Currently, Java enforces the output type to be the same as the two input type (way of least casts), but I'm not sure if we should relay on it.

So the first error is quite easily introduced when the signature for `toMillis()` changes (or a completely different method is used), and is near to impossible to see without debugging or tooltips.

The second error is just theoretical, since I personally saw this change happen (in python). There is also no way of figuring this error out, since the syntax in both version is correct. Nevertheless, it is safe to assume (I think @martinalig  has some input?) that the Java implementation will not in the near future change this behaviour.

So TL;DR my suggestion is to either use modulo calculus, or at least have a fixed type for rounding.

Reply:
I prefer fixed types of var anyway. So lets use the _long_ there. Besides that I do not see any issue here. Also we have some unittests which help us.